api/v1/lib/extras/offers/offers.go

```diff
@@ -0,0 +1,245 @@
+package offers
```

Comment:
(a) this is begging for type-generation
(b) offers are a scheduler concern, consider moving to `scheduler/offers`